reana_client/cli/workflow.py

```diff
@@ -729,22 +732,99 @@ def workflow_logs(ctx, workflow, access_token, json_format, steps=None):  # noqa
+        "slurm": "Slurm",
+    }
+    steps = []
+    chosen_filters = dict()
+
     logging.debug("command: {}".format(ctx.command_path.replace(" ", ".")))
     for p in ctx.params:
         logging.debug("{param}: {value}".format(param=p, value=ctx.params[p]))
     if workflow:
+        if filters:
+            try:
+                for f in filters:
+                    key, value = f.split("=")
+                    if key not in applicable_filters:
+                        click.echo(
+                            click.style(
+                                "Filter '{}' is not applicable. Applicable filters are {}".format(
+                                    key, list(applicable_filters.keys())
+                                ),
+                                fg="red",
+                            ),
+                            err=True,
+                        )
+                        return
+                    if key == "step":
+                        steps.append(value)
+                    else:
+                        # Case insensitive for compute backends
+                        value = (
+                            compute_backends[value.lower()]
+                            if key == "compute_backend"
+                            and value.lower() in compute_backends
+                            else value
+                        )
+                        chosen_filters[key] = value
+            except Exception as e:
+                logging.debug(traceback.format_exc())
+                logging.debug(str(e))
+                click.echo(
+                    click.style(
+                        "Cannot read filter from input '{}'. Please give it in the form key=value, e.g. status=failed\nApplicable filters are {}".format(
+                            f, list(applicable_filters.keys())
+                        ),
+                        fg="red",
+                    ),
+                    err=True,
+                )
+                return
         try:
-            if steps:
-                steps = list(set(steps))
-            response = get_workflow_logs(workflow, access_token, steps)
+            response = get_workflow_logs(
+                workflow, access_token, None if not steps else list(set(steps))
+            )
             workflow_logs = json.loads(response["logs"])
+            if filters:
+                for key, value in chosen_filters.items():
+                    unwanted_steps = [
+                        k
+                        for k, v in workflow_logs["job_logs"].items()
+                        if v[applicable_filters[key]] != value
+                    ]
+                    for job_id in unwanted_steps:
+                        del workflow_logs["job_logs"][job_id]
+
+                if not workflow_logs["job_logs"]:
+                    click.echo(
+                        click.style(
+                            "No matches found for {}\nApplicable filters are {}".format(
```

Comment:
(1) If the filter names are good, and if nothing is found, I think we can treat it as "filter passed successfully, but nothing came up after filter. 

```console
$ reana-client logs -w helloworld-yadage --filter status=failed
No matches found for ['step=failed'].
```

=> I'd say let's not print anything on the terminal in this case.  (Kind of like `docker ps` does not show anything when no container is running.)  This will allow people to use the commands in their own scripts (e.g.if they use `--json` output).

(2) If user provides wrong filter name, we should print error message and then sys.exit(1) so that exit status would indicate failure.

(3) The help messages could be prettified, for example:

```console
$ reana-client logs -w helloworld-yadage --filter steppp
Cannot read filter from input 'steppp'. Please give it in the form key=value, e.g. status=failed
```

=> Better say something like: "Error: please provide complete `--filter name=value` pairs, for example `--filter status=running`".

(4) You can also prettify messages like:

```console
$ reana-client logs -w helloworld-yadage --filter steppp=foo
Filter 'steppp' is not applicable. Applicable filters are ['step', 'compute_backend', 'docker_img', 'status']
```

=> Better said something like: "Error: filter `steppp` is not valid. Available filters are: 'compute_backend', 'docker_img', 'status', 'step'.

Reply:
I agree sounds better. Code updated!